### ISSUE TITLE
Support nested directories in dirs stanza

### ISSUE THREAD

**Issue by bandersongit** [ocaml/dune#1620]
Currently if I have the following directory structure
```
|  lib/
|      __tests__
|      dune
```
then `(dirs __tests__)` lets me pick up modules defined in __tests__

but if I instead have
```
|  project/
|         dune
|         lib/
|              __tests__
```
then `(dirs lib/__tests__)` does not let me pick up modules defined in __tests__

It would be great if there was support for the latter case

**Comment by TheAspiringHacker** [comment]
I'm not sure, but is [include_subdirs](https://dune.readthedocs.io/en/latest/dune-files.html#include-subdirs) what you're looking for?

**Comment by bandersongit** [comment]
No, my examples also had include_subdirs, the issue is with directories starting with `.` or `_` being excluded by default per the [dirs docs](https://dune.readthedocs.io/en/latest/dune-files.html#dirs-since-1-6). If I rename the `__tests__` directory to just `tests`, then it just works, but there are a couple reasons I can't just do that in my specific case.
